packages/account/src/account.ts

```diff
@@ -653,17 +657,39 @@ export class Account extends AbstractAccount implements WithAddress {
    */
   async sendTransaction(
     transactionRequestLike: TransactionRequestLike,
-    { estimateTxDependencies = true, onBeforeSend, skipCustomFee = false }: AccountSendTxParams = {}
+    { estimateTxDependencies = true, ...connectorOptions }: AccountSendTxParams = {}
   ): Promise<TransactionResponse> {
+    let transactionRequest = transactionRequestify(transactionRequestLike);
+
+    // Check if the account is using a connector, and therefore we do not have direct access to the
+    // private key.
     if (this._connector) {
-      return this.provider.getTransactionResponse(
-        await this._connector.sendTransaction(this.address.toString(), transactionRequestLike, {
-          onBeforeSend,
-          skipCustomFee,
-        })
+      const { onBeforeSend, skipCustomFee = false, data } = connectorOptions;
+
+      transactionRequest = await this.prepareTransactionForSend(transactionRequest);
+
+      const params: FuelConnectorSendTxParams = {
+        onBeforeSend,
+        skipCustomFee,
+        provider: {
+          url: this.provider.url,
+          cache: await serializeProviderCache(this.provider),
+        },
+        data,
+        state: transactionRequest.flag.state,
```

Comment:
```suggestion
        transactionState: transactionRequest.flag.state,
```

I think this makes more sense

Reply:
Yeah this makes more sense once it's passed through the connector as well.